cloudinit/sources/__init__.py

```diff
@@ -174,12 +174,20 @@ class DataSource(metaclass=abc.ABCMeta):
 
     # The datasource defines a set of supported EventTypes during which
     # the datasource can react to changes in metadata and regenerate
-    # network configuration on metadata changes.
-    # A datasource which supports writing network config on each system boot
-    # would call update_events['network'].add(EventType.BOOT).
+    # network configuration on metadata changes. These are defined in
+    # `supported_network_events`.
+    # The datasource also defines a set of default EventTypes that the
+    # datasource can react to. These are the event types that will be used
+    # if not overridden by the user.
+    # A datasource requiring to write network config on each system boot
+    # would call default_update_events['network'].add(EventType.BOOT).
```

Comment:
I know you didn't introduce this behaviour, but it's always made me a little concerned: we're modifying the set of events that _all_ datasource classes will use. This _shouldn't_ be a problem, because we only ever use one DS, but as your head is in this space ATM, let me know if you can think of any problems with it.

Reply:
Yeah, I'm not very comfortable with this either. I haven't thought of any easy solutions though.